### PR TITLE
Automate apmpackage publishing to package-storage (snapshot) 

### PR DESCRIPTION
**Pull request by v1v**
## Motivation/summary

Automate apmpackage publishing to package-storage (snapshot).

## How to test these changes

Run in the CI manually, the `packaging` pipeline doesn't get triggered automatically on a PR basis, then the UI is the only way to test this out.

## Implementation details

Makefile is the wrapper that runs `git` and `gh`.
Uses the `elastic/package-storage` repository, therefore branches are created directly in the upstream project.

## What 

* There are two actions that can trigger to publish a new apm package:
a. When identifying changes to the apmpackage after merging a PR to main or a release branch
b. When bumping the [version](https://github.com/elastic/apm-server/blob/master/cmd/version.go#L21) (usually happens after a stack release)
* Run make build-package, which builds the apm integration under the apm-server's root directory at build/integrations/apm/<version>.
* Run elastic-package check (https://github.com/elastic/elastic-package#elastic-package-check) on the created package and only continue if successful.
* Push the new apm package to the snapshot branch of the [package-storage](https://github.com/elastic/package-storage) repo. If a package with this version already exists, overwrite it, otherwise update the existing one.

## Related issues

Closes https://github.com/elastic/apm-server/issues/6970

## Steps

- [x] Create PR automatically if there are changes

## Test

See https://github.com/elastic/package-storage/pull/4095

## Follow ups 

- Once pushed, [release the changes to the elastic package registry](https://github.com/elastic/integrations/blob/master/docs/developer_workflow_promote_release_integration.md#steps) . Since it requires to validate if this PR is valid and then we can move forward to automate the release to Staging.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -237,6 +240,34 @@ $(PYTHON_BIN)/activate: $(MAGE) script/requirements.txt
...
+.PHONY: create-package-storage-pull-request
+create-package-storage-pull-request:
+	@cd $(PACKAGESTORAGE) ; git checkout -b update-apm-$(APM_SERVER_VERSION)-$(shell date "+%Y%m%d%H%M%S")
+	@cd $(PACKAGESTORAGE) ; git diff --staged --quiet || ( git add . ; git commit -m "[automation] Publish apm-$(APM_SERVER_VERSION)" )
+	@cd $(PACKAGESTORAGE) ; \
+		gh pr create \
+			--title "Publish apm-$(APM_SERVER_VERSION)" \
+			--body "Automated by $${BUILD_URL}" \
+			--label automation \
+			--assignee v1v \
```
Comment (v1v) [inline Makefile]: What team should be assigned?

**Review by cachedout — APPROVED** [review]

**Review by kuisathaverat — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -237,6 +241,43 @@ $(PYTHON_BIN)/activate: $(MAGE) script/requirements.txt
 $(APPROVALS):
 	@$(GO) build -o $@ github.com/elastic/apm-server/approvaltest/cmd/check-approvals
 
+##############################################################################
+# Rules for the package-storage.
+##############################################################################
```
Comment (axw) [inline Makefile:246]: Can we make these scripts in `.ci/scripts`? This is entirely CI-related, not something that developers will run directly.

**Review by axw — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (v1v) [reply Makefile]: I've just committed https://github.com/elastic/apm-server/pull/7900/commits/08fcd2f5146e68584e121179f51d61f3bdf27951 and tested it locally.

I'm gonna give a go in the CI and see how it goes

**Review by axw — APPROVED** [review]
thank you!

File: .ci/scripts/Makefile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+		git remote -v ; \
+		echo "INFO: push branch" ; \
+		git push --set-upstream origin $(PACKAGE_STORAGE_BRANCH) ; \
+		echo "INFO: create pull request" ; \
+		gh pr create \
+			--title "Publish apm-$(APM_SERVER_VERSION)" \
+			--body "Automated by $${BUILD_URL}" \
+			--label automation \
+			--base snapshot \
+			--head $(PACKAGE_STORAGE_BRANCH)
```
Comment (v1v) [inline .ci/scripts/Makefile:41]:
```suggestion
			--head $(PACKAGE_STORAGE_BRANCH) \
			--reviewer elastic/apm-server
```

**Comment by v1v** [comment]
unrelated failures

**Pull request merged**